src/shell.js

```diff
@@ -224,8 +255,58 @@ else if (ENVIRONMENT_IS_WEB || ENVIRONMENT_IS_WORKER) {
   }
 }
 else {
-  // Unreachable because SHELL is dependant on the others
-  throw 'Unknown runtime environment. Where are we?';
+  // Unreachable because SHELL is dependent on the others
+  throw new Error('Unknown runtime environment. Where are we?');
+}
+
+// Converts a string of base64 into a byte array.
+// Throws error on invalid input.
+function base64ToBytes(s) {
+  try {
+    var decoded = atob(s);
+    var bytes = new Uint8Array(decoded.length);
+    for (i = 0 ; i < decoded.length ; ++i) {
+      bytes[i] = decoded.charCodeAt(i);
+    }
+    return bytes;
+  } catch (_) {
+    throw new Error('Converting base64 string to bytes failed.');
+  }
+}
+
+// Converts a byte array into a string.
+// Returns invalid output on invalid input.
+function bytesToString(bytes) {
```

Comment:
Perfect, thanks. I just went ahead and moved all that stuff over to preamble.js and renamed `base64ToBytes` to `intArrayFromBase64` for consistency.